lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -955,6 +973,9 @@ static void printFModuleLikeOp(OpAsmPrinter &p, FModuleLike op) {
       "sym_name", "portDirections", "portTypes",  "portAnnotations",
       "portSyms", "portLocations",  "parameters", visibilityAttrName};
 
+  if (op.getConvention() == Convention::Internal)
+    omittedAttrs.push_back("convention");
```

Comment:
Consider increasing the SmallVector so it isn't resized if all optional are used.  Or push it entirely to the heap originally.